core/src/main/java/com/onelogin/saml2/util/Util.java

```diff
@@ -503,20 +504,19 @@ public static X509Certificate loadCert(String certString) throws CertificateExce
 	 * @return Loaded private key. PrivateKey object
 	 *
 	 * @throws GeneralSecurityException 
-	 * @throws IOException 
 	 */
-	public static PrivateKey loadPrivateKey(String keyString) throws GeneralSecurityException, IOException {
+	public static PrivateKey loadPrivateKey(String keyString) throws GeneralSecurityException {
 		org.apache.xml.security.Init.init();
 
-		keyString = formatPrivateKey(keyString, false);
-		keyString = chunkString(keyString, 64);		
+		String extractedKey = formatPrivateKey(keyString, false);
+		extractedKey = chunkString(extractedKey, 64);
 		KeyFactory kf = KeyFactory.getInstance("RSA");
 		
 		PrivateKey privKey;
 		try {
-			byte[] encoded = Base64.decodeBase64(keyString);
+			byte[] encoded = Base64.decodeBase64(extractedKey);
 			PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(encoded);
-			privKey = (PrivateKey) kf.generatePrivate(keySpec);
+			privKey = kf.generatePrivate(keySpec);
```

Comment:
PrivateKey cast here is not required?

Reply:
No. The KeyFactory#generatePrivate method returns already a PrivateKey